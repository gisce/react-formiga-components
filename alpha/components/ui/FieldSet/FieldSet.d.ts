import React from "react";
import { FieldSetProps } from "./FieldSet.types";
export declare const FieldSet: (props: FieldSetProps) => React.ReactElement;
//# sourceMappingURL=FieldSet.d.ts.map