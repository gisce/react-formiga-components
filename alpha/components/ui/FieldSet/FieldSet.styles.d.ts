export declare const FieldSetElement: import("styled-components").StyledComponent<"fieldset", any, {}, never>;
export declare const Legend: import("styled-components").StyledComponent<"legend", any, {}, never>;
//# sourceMappingURL=FieldSet.styles.d.ts.map