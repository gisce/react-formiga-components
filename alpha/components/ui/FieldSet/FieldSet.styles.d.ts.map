{"version":3,"file":"FieldSet.styles.d.ts","sourceRoot":"","sources":["../../../../src/components/ui/FieldSet/FieldSet.styles.tsx"],"names":[],"mappings":"AAEA,eAAO,MAAM,eAAe,yEAQ3B,CAAC;AACF,eAAO,MAAM,MAAM,uEAOlB,CAAC"}