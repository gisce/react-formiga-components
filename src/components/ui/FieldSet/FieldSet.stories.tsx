import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { FieldSet } from "./FieldSet";

export default {
  title: "Components/UI/FieldSet",
  component: FieldSet,
  argTypes: {
    children: { control: { disable: true } },
  },
} as ComponentMeta<typeof FieldSet>;

const Template: ComponentStory<typeof FieldSet> = (args) => {
  return <FieldSet {...args} />;
};

export const Basic = Template.bind({});
Basic.args = {
  label: "Group of fields",
  children: <>Content</>,
};
