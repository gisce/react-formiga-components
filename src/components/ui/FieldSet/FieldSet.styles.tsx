import styled from "styled-components";

export const FieldSetElement = styled.fieldset<{ borderRadius?: string }>`
  padding: 0.5rem;
  border-color: #d9d9d9;
  border-width: 1px;
  border-style: solid;
  border-radius: ${(props) => props.borderRadius || "5px"};
  font-family:
    ui-sans-serif,
    system-ui,
    -apple-system,
    BlinkMacSystemFont,
    "Segoe UI",
    Roboto,
    "Helvetica Neue",
    Arial,
    "Noto Sans",
    sans-serif,
    "Apple Color Emoji",
    "Segoe UI Emoji",
    "Segoe UI Symbol",
    "Noto Color Emoji" !important;
`;

export const Legend = styled.legend`
  all: initial;
  padding-left: 0.5em;
  padding-right: 0.5em;
  font-family:
    system-ui,
    -apple-system,
    "Segoe UI",
    Roboto,
    Helvetica,
    Arial,
    sans-serif,
    "Apple Color Emoji",
    "Segoe UI Emoji";
  font-size: 0.85rem;
`;
