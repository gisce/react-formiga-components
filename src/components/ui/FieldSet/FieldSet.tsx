import React from "react";
import { FieldSetElement, Legend } from "./FieldSet.styles";
import { FieldSetProps } from "./FieldSet.types";

export const FieldSet = (props: FieldSetProps): React.ReactElement => {
  const { label, children } = props;

  return (
    <FieldSetElement>
      <Legend>{label}</Legend>
      {children}
    </FieldSetElement>
  );
};
