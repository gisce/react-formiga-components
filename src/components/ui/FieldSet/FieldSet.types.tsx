export type FieldSetProps = {
  label: string;
  children: any;
};
