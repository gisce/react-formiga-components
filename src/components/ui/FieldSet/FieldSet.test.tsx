import React from "react";
import { render } from "@testing-library/react";
import { composeStories } from "@storybook/testing-react";
import * as stories from "./FieldSet.stories";
import matchMediaMock from "@/mocks/matchMedia.mock";

describe("FieldSet", () => {
  beforeAll(() => {
    matchMediaMock();
  });

  const testCases = Object.values(composeStories(stories)).map((Story) => [
    Story.storyName!,
    Story,
  ]);

  test.each(testCases)("Renders %s story", async (_storyName, Story) => {
    const tree = render(<Story />);
    expect(tree.baseElement).toMatchSnapshot();
  });
});
