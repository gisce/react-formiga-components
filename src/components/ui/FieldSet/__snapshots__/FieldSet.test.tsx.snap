// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldSet Renders Basic story 1`] = `
.c0 {
  padding: 0.5rem;
  border-color: rga(161,161,170);
  border-width: 1px;
  border-style: solid;
  font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont, "Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif, "Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji" !important;
}

.c1 {
  all: initial;
  margin-left: 1em;
  padding-left: 0.5em;
  padding-right: 0.5em;
  font-family: system-ui,-apple-system,"Segoe UI",Roboto,Helvetica,Arial, sans-serif,"Apple Color Emoji","Segoe UI Emoji";
}

<body>
  <div>
    <fieldset
      class="c0"
    >
      <legend
        class="c1"
      >
        Group of fields
      </legend>
      Content
    </fieldset>
  </div>
</body>
`;
